### PR TITLE
veracrypt: init at 1.19

### PR DESCRIPTION
**Pull request by dsferruzza**
###### Motivation for this change

[TrueCrypt](https://en.wikipedia.org/wiki/TrueCrypt) has been discontinued for several years.
This PR adds a package for [VeraCrypt](https://en.wikipedia.org/wiki/VeraCrypt), a maintained fork of TrueCrypt.

I used `truecrypt` [package](https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/misc/truecrypt/default.nix) as a start, and tried to create a nice package; but I'm a beginner so I guess not everything is perfect.
I would be happy to have feedback and advices!

I tested this package on NixOS.
I was able to successfully do the following steps:
- run VeraCrypt
- create a container
- mount it
- store a file in it
- unmount it
- move it to another computer running another OS with an official binary version of VeraCrypt
- mount it
- recover my file

I let the `maintainers` field empty.
I don't know if I should add myself, or @viric (the `truecrypt` package maintainer), or someone else.

###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/misc/veracrypt/default.nix
Comment (ndowens) [inline pkgs/applications/misc/veracrypt/default.nix]: pkgconfig goes into nativeBuildInputs

File: pkgs/applications/misc/veracrypt/default.nix
Comment (ndowens) [inline pkgs/applications/misc/veracrypt/default.nix]: Should be platforms = platforms.linux since it is only one platform

**Review by ndowens — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/veracrypt/default.nix
Comment (dsferruzza) [inline pkgs/applications/misc/veracrypt/default.nix]: Done!

File: pkgs/applications/misc/veracrypt/default.nix
Comment (dsferruzza) [inline pkgs/applications/misc/veracrypt/default.nix]: Done!

**Review by ndowens — APPROVED** [review]

**Comment by dsferruzza** [comment]
It seems the first change broke the build: https://travis-ci.org/NixOS/nixpkgs/jobs/216476198#L1575

@ndowens Any idea of why it happened?
I tried it on my machine but it didn't rebuild the package so I assumed it didn't need to because the result would be the same.
Also I couldn't find the difference between `buildInputs` and `nativeBuildInputs`.

File: pkgs/applications/misc/veracrypt/default.nix
Comment (Mic92) [inline pkgs/applications/misc/veracrypt/default.nix]: Do you want to maintains this package?

**Comment by ndowens** [comment]
The change didn't break it. Travis seems to be broken again. The difference between the buildinputs is that native is is used for building the pkg vs buildinputs is for what the pkg uses for runtime

File: pkgs/applications/misc/veracrypt/default.nix
Comment (dsferruzza) [inline pkgs/applications/misc/veracrypt/default.nix]: What does it implies exactly?
Updating the package when there is an update upstream?

**Comment by dsferruzza** [comment]
@ndowens I don't believe that the names perfectly describe the semantics, but thanks for the explanation I'll keep that in mind :)

File: pkgs/applications/misc/veracrypt/default.nix
Comment (joachifm) [inline pkgs/applications/misc/veracrypt/default.nix]: Typo: should be `nativeBuildInputs`

**Comment by joachifm** [comment]
The travis error is legit.

File: pkgs/applications/misc/veracrypt/default.nix
Comment (dsferruzza) [inline pkgs/applications/misc/veracrypt/default.nix]: Oh indeed! I fixed it

File: pkgs/applications/misc/veracrypt/default.nix
Comment (joachifm) [inline pkgs/applications/misc/veracrypt/default.nix]: It implies that you have some vested interest in the package; I don't think there's an implication of a certain level of commitment, really, but you can expect to be pinged if there's an issue.

Comment (dsferruzza) [inline pkgs/applications/misc/veracrypt/default.nix]: OK I understand!
So I shall add myself in `lib/maintainers.nix` and update the field in the package, right?

Comment (joachifm) [inline pkgs/applications/misc/veracrypt/default.nix]: That'd be great :)

Comment (dsferruzza) [inline pkgs/applications/misc/veracrypt/default.nix]: Done!

**Comment by joachifm** [comment]
Added a few things for you to consider.

**Comment by dsferruzza** [comment]
Thanks I learned a few tips ^_^

**Comment by joachifm** [comment]
Thank you

**Comment by dsferruzza** [comment]
Thank you for helping me ^_^

Do you now if/when the package will be available in a release channel?

**Comment by joachifm** [comment]
It's already in the unstable channel, but hasn't been picked to the release branch.

**Comment by dsferruzza** [comment]
OK!
So how does it work? Do I need to wait until the next release freeze or is there another way?

**Comment by joachifm** [comment]
You can submit a PR against the release branch, containing cherry-picked patches (please use `-x` to retain a reference to the original commit).

**Pull request merged**
